internal/pkg/gateway/commit/channelnotifier.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2021 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit
+
+import (
+	"sync"
+
+	"github.com/hyperledger/fabric/core/ledger"
+)
+
+type channelNotifier struct {
+	lock          sync.Mutex
+	commitChannel <-chan *ledger.CommitNotification
+	listeners     map[*channelListener]bool
```

Comment:
Using a pointer as a key is very risky. Why are we doing this?

Reply:
Outdated